// lilt-node/dist/model/Comment 2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Annotation = _interopRequireDefault(require("./Annotation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Comment model module.
 * @module model/Comment
 * @version 0.5.0
 */
var Comment = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Comment</code>.
   * A Comment is a translator&#39;s or a reviewer&#39;s comment on a segment. 
   * @alias module:model/Comment
   */
  function Comment() {
    _classCallCheck(this, Comment);

    Comment.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Comment, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Comment</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Comment} obj Optional instance to populate.
     * @return {module:model/Comment} The populated <code>Comment</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Comment();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }

        if (data.hasOwnProperty('text')) {
          obj['text'] = _ApiClient["default"].convertToType(data['text'], 'String');
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
        }

        if (data.hasOwnProperty('is_resolved')) {
          obj['is_resolved'] = _ApiClient["default"].convertToType(data['is_resolved'], 'Boolean');
        }

        if (data.hasOwnProperty('annotations')) {
          obj['annotations'] = _ApiClient["default"].convertToType(data['annotations'], [_Annotation["default"]]);
        }

        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return Comment;
}();
/**
 * A unique number identifying the Comment.
 * @member {Number} id
 */


Comment.prototype['id'] = undefined;
/**
 * The comment text.
 * @member {String} text
 */

Comment.prototype['text'] = undefined;
/**
 * The User who created this Comment.
 * @member {Number} user_id
 */

Comment.prototype['user_id'] = undefined;
/**
 * Whether the Comment is resolved.
 * @member {Boolean} is_resolved
 */

Comment.prototype['is_resolved'] = undefined;
/**
 * A list of optional Annotations.
 * @member {Array.<module:model/Annotation>} annotations
 */

Comment.prototype['annotations'] = undefined;
/**
 * Time at which the object was created. Measured in seconds since the Unix epoch.
 * @member {Number} created_at
 */

Comment.prototype['created_at'] = undefined;
var _default = Comment;
exports["default"] = _default;
