// lilt-node/dist/model/ConnectorArguments 2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ConnectorArguments model module.
 * @module model/ConnectorArguments
 * @version 0.5.0
 */
var ConnectorArguments = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConnectorArguments</code>.
   * @alias module:model/ConnectorArguments
   */
  function ConnectorArguments() {
    _classCallCheck(this, ConnectorArguments);

    ConnectorArguments.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ConnectorArguments, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ConnectorArguments</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ConnectorArguments} obj Optional instance to populate.
     * @return {module:model/ConnectorArguments} The populated <code>ConnectorArguments</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConnectorArguments();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('args')) {
          obj['args'] = _ApiClient["default"].convertToType(data['args'], Object);
        }

        if (data.hasOwnProperty('schedule')) {
          obj['schedule'] = _ApiClient["default"].convertToType(data['schedule'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ConnectorArguments;
}();
/**
 * A unique Connector identifier.
 * @member {Number} id
 */


ConnectorArguments.prototype['id'] = undefined;
/**
 * Name of connector.
 * @member {String} name
 */

ConnectorArguments.prototype['name'] = undefined;
/**
 * Connector parameters.
 * @member {Object} args
 */

ConnectorArguments.prototype['args'] = undefined;
/**
 * Cron string
 * @member {String} schedule
 */

ConnectorArguments.prototype['schedule'] = undefined;
var _default = ConnectorArguments;
exports["default"] = _default;
