// lilt-node/dist/model/DocumentPretranslateResponse 2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DocumentPretranslating = _interopRequireDefault(require("./DocumentPretranslating"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DocumentPretranslateResponse model module.
 * @module model/DocumentPretranslateResponse
 * @version 0.5.0
 */
var DocumentPretranslateResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DocumentPretranslateResponse</code>.
   * @alias module:model/DocumentPretranslateResponse
   */
  function DocumentPretranslateResponse() {
    _classCallCheck(this, DocumentPretranslateResponse);

    DocumentPretranslateResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DocumentPretranslateResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DocumentPretranslateResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DocumentPretranslateResponse} obj Optional instance to populate.
     * @return {module:model/DocumentPretranslateResponse} The populated <code>DocumentPretranslateResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DocumentPretranslateResponse();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], ['Number']);
        }

        if (data.hasOwnProperty('is_pretranslating')) {
          obj['is_pretranslating'] = _ApiClient["default"].convertToType(data['is_pretranslating'], 'Boolean');
        }

        if (data.hasOwnProperty('documents')) {
          obj['documents'] = _ApiClient["default"].convertToType(data['documents'], [_DocumentPretranslating["default"]]);
        }
      }

      return obj;
    }
  }]);

  return DocumentPretranslateResponse;
}();
/**
 * A list of documents being pretranslated.
 * @member {Array.<Number>} id
 */


DocumentPretranslateResponse.prototype['id'] = undefined;
/**
 * If the operation succeeded, then `true`. Otherwise, `false`.
 * @member {Boolean} is_pretranslating
 */

DocumentPretranslateResponse.prototype['is_pretranslating'] = undefined;
/**
 * Document pretranslation status.
 * @member {Array.<module:model/DocumentPretranslating>} documents
 */

DocumentPretranslateResponse.prototype['documents'] = undefined;
var _default = DocumentPretranslateResponse;
exports["default"] = _default;
