// lilt-node/dist/model/DocumentWithoutSegments.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DocumentWithoutSegmentsStatus = _interopRequireDefault(require("./DocumentWithoutSegmentsStatus"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DocumentWithoutSegments model module.
 * @module model/DocumentWithoutSegments
 * @version 0.5.0
 */
var DocumentWithoutSegments = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DocumentWithoutSegments</code>.
   * A Document is a collection of zero or more Segments. 
   * @alias module:model/DocumentWithoutSegments
   */
  function DocumentWithoutSegments() {
    _classCallCheck(this, DocumentWithoutSegments);

    DocumentWithoutSegments.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DocumentWithoutSegments, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DocumentWithoutSegments</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DocumentWithoutSegments} obj Optional instance to populate.
     * @return {module:model/DocumentWithoutSegments} The populated <code>DocumentWithoutSegments</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DocumentWithoutSegments();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }

        if (data.hasOwnProperty('project_id')) {
          obj['project_id'] = _ApiClient["default"].convertToType(data['project_id'], 'Number');
        }

        if (data.hasOwnProperty('srclang')) {
          obj['srclang'] = _ApiClient["default"].convertToType(data['srclang'], 'String');
        }

        if (data.hasOwnProperty('trglang')) {
          obj['trglang'] = _ApiClient["default"].convertToType(data['trglang'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('import_in_progress')) {
          obj['import_in_progress'] = _ApiClient["default"].convertToType(data['import_in_progress'], 'Boolean');
        }

        if (data.hasOwnProperty('import_succeeded')) {
          obj['import_succeeded'] = _ApiClient["default"].convertToType(data['import_succeeded'], 'Boolean');
        }

        if (data.hasOwnProperty('import_error_message')) {
          obj['import_error_message'] = _ApiClient["default"].convertToType(data['import_error_message'], 'String');
        }

        if (data.hasOwnProperty('export_in_progress')) {
          obj['export_in_progress'] = _ApiClient["default"].convertToType(data['export_in_progress'], 'Boolean');
        }

        if (data.hasOwnProperty('export_succeeded')) {
          obj['export_succeeded'] = _ApiClient["default"].convertToType(data['export_succeeded'], 'Boolean');
        }

        if (data.hasOwnProperty('export_error_message')) {
          obj['export_error_message'] = _ApiClient["default"].convertToType(data['export_error_message'], 'String');
        }

        if (data.hasOwnProperty('is_pretranslating')) {
          obj['is_pretranslating'] = _ApiClient["default"].convertToType(data['is_pretranslating'], 'Boolean');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _DocumentWithoutSegmentsStatus["default"].constructFromObject(data['status']);
        }

        if (data.hasOwnProperty('translator_email')) {
          obj['translator_email'] = _ApiClient["default"].convertToType(data['translator_email'], 'String');
        }

        if (data.hasOwnProperty('reviewer_email')) {
          obj['reviewer_email'] = _ApiClient["default"].convertToType(data['reviewer_email'], 'String');
        }

        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Number');
        }

        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return DocumentWithoutSegments;
}();
/**
 * A unique number identifying the Document.
 * @member {Number} id
 */


DocumentWithoutSegments.prototype['id'] = undefined;
/**
 * A unique number identifying the Project.
 * @member {Number} project_id
 */

DocumentWithoutSegments.prototype['project_id'] = undefined;
/**
 * An ISO 639-1 language identifier.
 * @member {String} srclang
 */

DocumentWithoutSegments.prototype['srclang'] = undefined;
/**
 * An ISO 639-1 language identifier.
 * @member {String} trglang
 */

DocumentWithoutSegments.prototype['trglang'] = undefined;
/**
 * The document name.
 * @member {String} name
 */

DocumentWithoutSegments.prototype['name'] = undefined;
/**
 * True if the document is currently being imported
 * @member {Boolean} import_in_progress
 */

DocumentWithoutSegments.prototype['import_in_progress'] = undefined;
/**
 * True if the import process succeeded.
 * @member {Boolean} import_succeeded
 */

DocumentWithoutSegments.prototype['import_succeeded'] = undefined;
/**
 * Error message if `import_succeeded=false`
 * @member {String} import_error_message
 */

DocumentWithoutSegments.prototype['import_error_message'] = undefined;
/**
 * True if the document is currently being exported for download
 * @member {Boolean} export_in_progress
 */

DocumentWithoutSegments.prototype['export_in_progress'] = undefined;
/**
 * True if the export process succeeded.
 * @member {Boolean} export_succeeded
 */

DocumentWithoutSegments.prototype['export_succeeded'] = undefined;
/**
 * Error message if `export_succeeded=false`
 * @member {String} export_error_message
 */

DocumentWithoutSegments.prototype['export_error_message'] = undefined;
/**
 * True if the document is currently being pretranslated.
 * @member {Boolean} is_pretranslating
 */

DocumentWithoutSegments.prototype['is_pretranslating'] = undefined;
/**
 * @member {module:model/DocumentWithoutSegmentsStatus} status
 */

DocumentWithoutSegments.prototype['status'] = undefined;
/**
 * The email of the assigned translator.
 * @member {String} translator_email
 */

DocumentWithoutSegments.prototype['translator_email'] = undefined;
/**
 * The email of the assigned reviewer.
 * @member {String} reviewer_email
 */

DocumentWithoutSegments.prototype['reviewer_email'] = undefined;
/**
 * Time at which the object was created. Measured in seconds since the Unix epoch.
 * @member {Number} created_at
 */

DocumentWithoutSegments.prototype['created_at'] = undefined;
/**
 * Time at which the object was created. Measured in seconds since the Unix epoch.
 * @member {Number} updated_at
 */

DocumentWithoutSegments.prototype['updated_at'] = undefined;
var _default = DocumentWithoutSegments;
exports["default"] = _default;
