// lilt-node/dist/model/DocumentPretranslating.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DocumentPretranslatingStatus = _interopRequireDefault(require("./DocumentPretranslatingStatus"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DocumentPretranslating model module.
 * @module model/DocumentPretranslating
 * @version 0.5.0
 */
var DocumentPretranslating = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DocumentPretranslating</code>.
   * Document object for which pretranslation was requested. 
   * @alias module:model/DocumentPretranslating
   */
  function DocumentPretranslating() {
    _classCallCheck(this, DocumentPretranslating);

    DocumentPretranslating.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DocumentPretranslating, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DocumentPretranslating</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DocumentPretranslating} obj Optional instance to populate.
     * @return {module:model/DocumentPretranslating} The populated <code>DocumentPretranslating</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DocumentPretranslating();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }

        if (data.hasOwnProperty('import_in_progress')) {
          obj['import_in_progress'] = _ApiClient["default"].convertToType(data['import_in_progress'], 'Boolean');
        }

        if (data.hasOwnProperty('import_succeeded')) {
          obj['import_succeeded'] = _ApiClient["default"].convertToType(data['import_succeeded'], 'Boolean');
        }

        if (data.hasOwnProperty('import_error_message')) {
          obj['import_error_message'] = _ApiClient["default"].convertToType(data['import_error_message'], 'String');
        }

        if (data.hasOwnProperty('is_processing')) {
          obj['is_processing'] = _ApiClient["default"].convertToType(data['is_processing'], 'Boolean');
        }

        if (data.hasOwnProperty('is_pretranslating')) {
          obj['is_pretranslating'] = _ApiClient["default"].convertToType(data['is_pretranslating'], 'Boolean');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _DocumentPretranslatingStatus["default"].constructFromObject(data['status']);
        }
      }

      return obj;
    }
  }]);

  return DocumentPretranslating;
}();
/**
 * A status object indicating the pretranslation status.
 * @member {Number} id
 */


DocumentPretranslating.prototype['id'] = undefined;
/**
 * Indicates that the document is being imported.
 * @member {Boolean} import_in_progress
 */

DocumentPretranslating.prototype['import_in_progress'] = undefined;
/**
 * Indicates that the document was successfully imported.
 * @member {Boolean} import_succeeded
 */

DocumentPretranslating.prototype['import_succeeded'] = undefined;
/**
 * Indicates there was an error importing the document.
 * @member {String} import_error_message
 */

DocumentPretranslating.prototype['import_error_message'] = undefined;
/**
 * Indicates the document is being processed.
 * @member {Boolean} is_processing
 */

DocumentPretranslating.prototype['is_processing'] = undefined;
/**
 * Indicates the document is being pretranslated.
 * @member {Boolean} is_pretranslating
 */

DocumentPretranslating.prototype['is_pretranslating'] = undefined;
/**
 * @member {module:model/DocumentPretranslatingStatus} status
 */

DocumentPretranslating.prototype['status'] = undefined;
var _default = DocumentPretranslating;
exports["default"] = _default;
