// lilt-node/dist/model/File 2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The File model module.
 * @module model/File
 * @version 0.5.0
 */
var File = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>File</code>.
   * A File is an unprocessed source file that can later be added to a project.
   * @alias module:model/File
   */
  function File() {
    _classCallCheck(this, File);

    File.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(File, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>File</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/File} obj Optional instance to populate.
     * @return {module:model/File} The populated <code>File</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new File();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('file_hash')) {
          obj['file_hash'] = _ApiClient["default"].convertToType(data['file_hash'], 'String');
        }

        if (data.hasOwnProperty('export_uri')) {
          obj['export_uri'] = _ApiClient["default"].convertToType(data['export_uri'], 'String');
        }

        if (data.hasOwnProperty('detected_lang')) {
          obj['detected_lang'] = _ApiClient["default"].convertToType(data['detected_lang'], 'String');
        }

        if (data.hasOwnProperty('detected_lang_confidence')) {
          obj['detected_lang_confidence'] = _ApiClient["default"].convertToType(data['detected_lang_confidence'], 'Number');
        }

        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Number');
        }

        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return File;
}();
/**
 * A unique number identifying the File.
 * @member {Number} id
 */


File.prototype['id'] = undefined;
/**
 * The file name.
 * @member {String} name
 */

File.prototype['name'] = undefined;
/**
 * A unique hash value associated with the file. An MD5 hash of the file content will be used by default.
 * @member {String} file_hash
 */

File.prototype['file_hash'] = undefined;
/**
 * A webhook endpoint that will export the translated document back to the source repository.
 * @member {String} export_uri
 */

File.prototype['export_uri'] = undefined;
/**
 * Language associated with the file.
 * @member {String} detected_lang
 */

File.prototype['detected_lang'] = undefined;
/**
 * Confidence score for the language associated with the file.
 * @member {Number} detected_lang_confidence
 */

File.prototype['detected_lang_confidence'] = undefined;
/**
 * Time at which the object was created. Measured in seconds since the Unix epoch.
 * @member {Number} created_at
 */

File.prototype['created_at'] = undefined;
/**
 * Time at which the object was created. Measured in seconds since the Unix epoch.
 * @member {Number} updated_at
 */

File.prototype['updated_at'] = undefined;
var _default = File;
exports["default"] = _default;
