// lilt-node/dist/model/MatchBand 2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The MatchBand model module.
 * @module model/MatchBand
 * @version 0.5.0
 */
var MatchBand = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MatchBand</code>.
   * A translation memory match band. 
   * @alias module:model/MatchBand
   */
  function MatchBand() {
    _classCallCheck(this, MatchBand);

    MatchBand.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MatchBand, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MatchBand</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MatchBand} obj Optional instance to populate.
     * @return {module:model/MatchBand} The populated <code>MatchBand</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MatchBand();

        if (data.hasOwnProperty('minimum_score')) {
          obj['minimum_score'] = _ApiClient["default"].convertToType(data['minimum_score'], 'Number');
        }

        if (data.hasOwnProperty('maximum_score')) {
          obj['maximum_score'] = _ApiClient["default"].convertToType(data['maximum_score'], 'Number');
        }

        if (data.hasOwnProperty('num_source_words')) {
          obj['num_source_words'] = _ApiClient["default"].convertToType(data['num_source_words'], 'Number');
        }

        if (data.hasOwnProperty('num_segments')) {
          obj['num_segments'] = _ApiClient["default"].convertToType(data['num_segments'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return MatchBand;
}();
/**
 * The minimum fuzzy match score.
 * @member {Number} minimum_score
 */


MatchBand.prototype['minimum_score'] = undefined;
/**
 * The maximum fuzzy match score.
 * @member {Number} maximum_score
 */

MatchBand.prototype['maximum_score'] = undefined;
/**
 * The number of source words in the band.
 * @member {Number} num_source_words
 */

MatchBand.prototype['num_source_words'] = undefined;
/**
 * The number of source segments in the band.
 * @member {Number} num_segments
 */

MatchBand.prototype['num_segments'] = undefined;
var _default = MatchBand;
exports["default"] = _default;
