// lilt-node/dist/model/MemoryCreateParameters.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The MemoryCreateParameters model module.
 * @module model/MemoryCreateParameters
 * @version 0.5.0
 */
var MemoryCreateParameters = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MemoryCreateParameters</code>.
   * @alias module:model/MemoryCreateParameters
   * @param name {String} A name for the Memory.
   * @param srclang {String} An ISO 639-1 language identifier.
   * @param trglang {String} An ISO 639-1 language identifier.
   */
  function MemoryCreateParameters(name, srclang, trglang) {
    _classCallCheck(this, MemoryCreateParameters);

    MemoryCreateParameters.initialize(this, name, srclang, trglang);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MemoryCreateParameters, null, [{
    key: "initialize",
    value: function initialize(obj, name, srclang, trglang) {
      obj['name'] = name;
      obj['srclang'] = srclang;
      obj['trglang'] = trglang;
    }
    /**
     * Constructs a <code>MemoryCreateParameters</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MemoryCreateParameters} obj Optional instance to populate.
     * @return {module:model/MemoryCreateParameters} The populated <code>MemoryCreateParameters</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MemoryCreateParameters();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('srclang')) {
          obj['srclang'] = _ApiClient["default"].convertToType(data['srclang'], 'String');
        }

        if (data.hasOwnProperty('trglang')) {
          obj['trglang'] = _ApiClient["default"].convertToType(data['trglang'], 'String');
        }

        if (data.hasOwnProperty('srclocale')) {
          obj['srclocale'] = _ApiClient["default"].convertToType(data['srclocale'], 'String');
        }

        if (data.hasOwnProperty('trglocale')) {
          obj['trglocale'] = _ApiClient["default"].convertToType(data['trglocale'], 'String');
        }
      }

      return obj;
    }
  }]);

  return MemoryCreateParameters;
}();
/**
 * A name for the Memory.
 * @member {String} name
 */


MemoryCreateParameters.prototype['name'] = undefined;
/**
 * An ISO 639-1 language identifier.
 * @member {String} srclang
 */

MemoryCreateParameters.prototype['srclang'] = undefined;
/**
 * An ISO 639-1 language identifier.
 * @member {String} trglang
 */

MemoryCreateParameters.prototype['trglang'] = undefined;
/**
 * An ISO 3166-1 region name for language locales
 * @member {String} srclocale
 */

MemoryCreateParameters.prototype['srclocale'] = undefined;
/**
 * An ISO 3166-1 region name for language locales
 * @member {String} trglocale
 */

MemoryCreateParameters.prototype['trglocale'] = undefined;
var _default = MemoryCreateParameters;
exports["default"] = _default;
