// lilt-node/dist/model/ProjectQuote.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _DocumentQuote = _interopRequireDefault(require("./DocumentQuote"));

var _MatchBand = _interopRequireDefault(require("./MatchBand"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ProjectQuote model module.
 * @module model/ProjectQuote
 * @version 0.5.0
 */
var ProjectQuote = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProjectQuote</code>.
   * Quoting information for a Project. 
   * @alias module:model/ProjectQuote
   */
  function ProjectQuote() {
    _classCallCheck(this, ProjectQuote);

    ProjectQuote.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ProjectQuote, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ProjectQuote</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProjectQuote} obj Optional instance to populate.
     * @return {module:model/ProjectQuote} The populated <code>ProjectQuote</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProjectQuote();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }

        if (data.hasOwnProperty('num_source_words')) {
          obj['num_source_words'] = _ApiClient["default"].convertToType(data['num_source_words'], 'Number');
        }

        if (data.hasOwnProperty('num_words_new')) {
          obj['num_words_new'] = _ApiClient["default"].convertToType(data['num_words_new'], 'Number');
        }

        if (data.hasOwnProperty('num_segments_new')) {
          obj['num_segments_new'] = _ApiClient["default"].convertToType(data['num_segments_new'], 'Number');
        }

        if (data.hasOwnProperty('num_words_repetition')) {
          obj['num_words_repetition'] = _ApiClient["default"].convertToType(data['num_words_repetition'], 'Number');
        }

        if (data.hasOwnProperty('num_segments_repetition')) {
          obj['num_segments_repetition'] = _ApiClient["default"].convertToType(data['num_segments_repetition'], 'Number');
        }

        if (data.hasOwnProperty('bands')) {
          obj['bands'] = _ApiClient["default"].convertToType(data['bands'], [_MatchBand["default"]]);
        }

        if (data.hasOwnProperty('documents')) {
          obj['documents'] = _ApiClient["default"].convertToType(data['documents'], [_DocumentQuote["default"]]);
        }
      }

      return obj;
    }
  }]);

  return ProjectQuote;
}();
/**
 * A unique Project identifier.
 * @member {Number} id
 */


ProjectQuote.prototype['id'] = undefined;
/**
 * The number of source words in the Project.
 * @member {Number} num_source_words
 */

ProjectQuote.prototype['num_source_words'] = undefined;
/**
 * The number of new source words in the Project.
 * @member {Number} num_words_new
 */

ProjectQuote.prototype['num_words_new'] = undefined;
/**
 * The number of new segments in the Project.
 * @member {Number} num_segments_new
 */

ProjectQuote.prototype['num_segments_new'] = undefined;
/**
 * The number of repetition source words in the Project.
 * @member {Number} num_words_repetition
 */

ProjectQuote.prototype['num_words_repetition'] = undefined;
/**
 * The number of repetition segments in the Project.
 * @member {Number} num_segments_repetition
 */

ProjectQuote.prototype['num_segments_repetition'] = undefined;
/**
 * A list of MatchBand objects that represent translation memory leverage statistics.
 * @member {Array.<module:model/MatchBand>} bands
 */

ProjectQuote.prototype['bands'] = undefined;
/**
 * A list of DocumentQuote objects that quotes information for a Document.
 * @member {Array.<module:model/DocumentQuote>} documents
 */

ProjectQuote.prototype['documents'] = undefined;
var _default = ProjectQuote;
exports["default"] = _default;
