// lilt-node/dist/model/ProjectUpdateResponse 2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ProjectUpdateResponse model module.
 * @module model/ProjectUpdateResponse
 * @version 0.5.0
 */
var ProjectUpdateResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProjectUpdateResponse</code>.
   * @alias module:model/ProjectUpdateResponse
   * @param id {Number} A unique Project identifier.
   */
  function ProjectUpdateResponse(id) {
    _classCallCheck(this, ProjectUpdateResponse);

    ProjectUpdateResponse.initialize(this, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ProjectUpdateResponse, null, [{
    key: "initialize",
    value: function initialize(obj, id) {
      obj['id'] = id;
    }
    /**
     * Constructs a <code>ProjectUpdateResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProjectUpdateResponse} obj Optional instance to populate.
     * @return {module:model/ProjectUpdateResponse} The populated <code>ProjectUpdateResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProjectUpdateResponse();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('state')) {
          obj['state'] = _ApiClient["default"].convertToType(data['state'], 'String');
        }

        if (data.hasOwnProperty('due_date')) {
          obj['due_date'] = _ApiClient["default"].convertToType(data['due_date'], 'Number');
        }

        if (data.hasOwnProperty('archived')) {
          obj['archived'] = _ApiClient["default"].convertToType(data['archived'], 'Boolean');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], Object);
        }

        if (data.hasOwnProperty('sample_review_percentage')) {
          obj['sample_review_percentage'] = _ApiClient["default"].convertToType(data['sample_review_percentage'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return ProjectUpdateResponse;
}();
/**
 * A unique Project identifier.
 * @member {Number} id
 */


ProjectUpdateResponse.prototype['id'] = undefined;
/**
 * The Project name.
 * @member {String} name
 */

ProjectUpdateResponse.prototype['name'] = undefined;
/**
 * The project's state. The possible states are 'backlog', 'inProgress', 'inReview', 'inQA', and 'done'
 * @member {String} state
 */

ProjectUpdateResponse.prototype['state'] = undefined;
/**
 * The due date. Measured in seconds since the Unix epoch.
 * @member {Number} due_date
 */

ProjectUpdateResponse.prototype['due_date'] = undefined;
/**
 * True if the project is archived. Otherwise, false.
 * @member {Boolean} archived
 */

ProjectUpdateResponse.prototype['archived'] = undefined;
/**
 * Metadata associated with a project. This field must be valid JSON.
 * @member {Object} metadata
 */

ProjectUpdateResponse.prototype['metadata'] = undefined;
/**
 * The Project's sample review percentage. Must be an integer between 10 and 100, a multiple of 10 and greater than minimum value (displayed in error message).
 * @member {Number} sample_review_percentage
 */

ProjectUpdateResponse.prototype['sample_review_percentage'] = undefined;
var _default = ProjectUpdateResponse;
exports["default"] = _default;
