// lilt-node/dist/model/ProjectStatus.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ResourceStatus = _interopRequireDefault(require("./ResourceStatus"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ProjectStatus model module.
 * @module model/ProjectStatus
 * @version 0.5.0
 */
var ProjectStatus = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProjectStatus</code>.
   * The status of a Project. 
   * @alias module:model/ProjectStatus
   */
  function ProjectStatus() {
    _classCallCheck(this, ProjectStatus);

    ProjectStatus.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ProjectStatus, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ProjectStatus</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProjectStatus} obj Optional instance to populate.
     * @return {module:model/ProjectStatus} The populated <code>ProjectStatus</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProjectStatus();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }

        if (data.hasOwnProperty('num_source_words')) {
          obj['num_source_words'] = _ApiClient["default"].convertToType(data['num_source_words'], 'Number');
        }

        if (data.hasOwnProperty('num_words_confirmed')) {
          obj['num_words_confirmed'] = _ApiClient["default"].convertToType(data['num_words_confirmed'], 'Number');
        }

        if (data.hasOwnProperty('num_words_reviewed')) {
          obj['num_words_reviewed'] = _ApiClient["default"].convertToType(data['num_words_reviewed'], 'Number');
        }

        if (data.hasOwnProperty('time_elapsed')) {
          obj['time_elapsed'] = _ApiClient["default"].convertToType(data['time_elapsed'], 'Number');
        }

        if (data.hasOwnProperty('time_elapsed_translation')) {
          obj['time_elapsed_translation'] = _ApiClient["default"].convertToType(data['time_elapsed_translation'], 'Number');
        }

        if (data.hasOwnProperty('time_elapsed_research')) {
          obj['time_elapsed_research'] = _ApiClient["default"].convertToType(data['time_elapsed_research'], 'Number');
        }

        if (data.hasOwnProperty('time_elapsed_review')) {
          obj['time_elapsed_review'] = _ApiClient["default"].convertToType(data['time_elapsed_review'], 'Number');
        }

        if (data.hasOwnProperty('resources')) {
          obj['resources'] = _ApiClient["default"].convertToType(data['resources'], [_ResourceStatus["default"]]);
        }
      }

      return obj;
    }
  }]);

  return ProjectStatus;
}();
/**
 * A unique Project identifier.
 * @member {Number} id
 */


ProjectStatus.prototype['id'] = undefined;
/**
 * The number of source words in the Project.
 * @member {Number} num_source_words
 */

ProjectStatus.prototype['num_source_words'] = undefined;
/**
 * The number of confirmed source words.
 * @member {Number} num_words_confirmed
 */

ProjectStatus.prototype['num_words_confirmed'] = undefined;
/**
 * The number of reviewed source words.
 * @member {Number} num_words_reviewed
 */

ProjectStatus.prototype['num_words_reviewed'] = undefined;
/**
 * The total time spent on the project by all resources. Measured in milliseconds.
 * @member {Number} time_elapsed
 */

ProjectStatus.prototype['time_elapsed'] = undefined;
/**
 * The total time spent on translation by all resources. Measured in milliseconds.
 * @member {Number} time_elapsed_translation
 */

ProjectStatus.prototype['time_elapsed_translation'] = undefined;
/**
 * The total time spent on research by all resources. Measured in milliseconds.
 * @member {Number} time_elapsed_research
 */

ProjectStatus.prototype['time_elapsed_research'] = undefined;
/**
 * The total time spent on reviewing by all resources. Measured in milliseconds.
 * @member {Number} time_elapsed_review
 */

ProjectStatus.prototype['time_elapsed_review'] = undefined;
/**
 * A list of ResourceStatus objects that represent per-resource statistics.
 * @member {Array.<module:model/ResourceStatus>} resources
 */

ProjectStatus.prototype['resources'] = undefined;
var _default = ProjectStatus;
exports["default"] = _default;
