// lilt-node/dist/model/ProjectCreateParameters.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ProjectCreateParameters model module.
 * @module model/ProjectCreateParameters
 * @version 0.5.0
 */
var ProjectCreateParameters = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ProjectCreateParameters</code>.
   * @alias module:model/ProjectCreateParameters
   * @param name {String} A name for the Project.
   * @param memoryId {Number} The Memory to associate with this new Project.
   */
  function ProjectCreateParameters(name, memoryId) {
    _classCallCheck(this, ProjectCreateParameters);

    ProjectCreateParameters.initialize(this, name, memoryId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ProjectCreateParameters, null, [{
    key: "initialize",
    value: function initialize(obj, name, memoryId) {
      obj['name'] = name;
      obj['memory_id'] = memoryId;
    }
    /**
     * Constructs a <code>ProjectCreateParameters</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ProjectCreateParameters} obj Optional instance to populate.
     * @return {module:model/ProjectCreateParameters} The populated <code>ProjectCreateParameters</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ProjectCreateParameters();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('memory_id')) {
          obj['memory_id'] = _ApiClient["default"].convertToType(data['memory_id'], 'Number');
        }

        if (data.hasOwnProperty('file_ids')) {
          obj['file_ids'] = _ApiClient["default"].convertToType(data['file_ids'], ['Number']);
        }

        if (data.hasOwnProperty('due_date')) {
          obj['due_date'] = _ApiClient["default"].convertToType(data['due_date'], 'Number');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], Object);
        }
      }

      return obj;
    }
  }]);

  return ProjectCreateParameters;
}();
/**
 * A name for the Project.
 * @member {String} name
 */


ProjectCreateParameters.prototype['name'] = undefined;
/**
 * The Memory to associate with this new Project.
 * @member {Number} memory_id
 */

ProjectCreateParameters.prototype['memory_id'] = undefined;
/**
 * A list of Files to add to this new Project.
 * @member {Array.<Number>} file_ids
 */

ProjectCreateParameters.prototype['file_ids'] = undefined;
/**
 * The due date. Measured in seconds since the Unix epoch.
 * @member {Number} due_date
 */

ProjectCreateParameters.prototype['due_date'] = undefined;
/**
 * Metadata associated with a project. This field must be valid JSON.
 * @member {Object} metadata
 */

ProjectCreateParameters.prototype['metadata'] = undefined;
var _default = ProjectCreateParameters;
exports["default"] = _default;
