// lilt-node/dist/model/QARuleMatchesContext.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The QARuleMatchesContext model module.
 * @module model/QARuleMatchesContext
 * @version 0.5.0
 */
var QARuleMatchesContext = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>QARuleMatchesContext</code>.
   * @alias module:model/QARuleMatchesContext
   * @param length {Number} The length of the error in characters in the context.
   * @param offset {Number} The 0-based character offset of the error in the context text.
   * @param text {String} Context of the error, i.e. the error and some text to the left and to the left.
   */
  function QARuleMatchesContext(length, offset, text) {
    _classCallCheck(this, QARuleMatchesContext);

    QARuleMatchesContext.initialize(this, length, offset, text);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(QARuleMatchesContext, null, [{
    key: "initialize",
    value: function initialize(obj, length, offset, text) {
      obj['length'] = length;
      obj['offset'] = offset;
      obj['text'] = text;
    }
    /**
     * Constructs a <code>QARuleMatchesContext</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/QARuleMatchesContext} obj Optional instance to populate.
     * @return {module:model/QARuleMatchesContext} The populated <code>QARuleMatchesContext</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new QARuleMatchesContext();

        if (data.hasOwnProperty('length')) {
          obj['length'] = _ApiClient["default"].convertToType(data['length'], 'Number');
        }

        if (data.hasOwnProperty('offset')) {
          obj['offset'] = _ApiClient["default"].convertToType(data['offset'], 'Number');
        }

        if (data.hasOwnProperty('text')) {
          obj['text'] = _ApiClient["default"].convertToType(data['text'], 'String');
        }
      }

      return obj;
    }
  }]);

  return QARuleMatchesContext;
}();
/**
 * The length of the error in characters in the context.
 * @member {Number} length
 */


QARuleMatchesContext.prototype['length'] = undefined;
/**
 * The 0-based character offset of the error in the context text.
 * @member {Number} offset
 */

QARuleMatchesContext.prototype['offset'] = undefined;
/**
 * Context of the error, i.e. the error and some text to the left and to the left.
 * @member {String} text
 */

QARuleMatchesContext.prototype['text'] = undefined;
var _default = QARuleMatchesContext;
exports["default"] = _default;
