// lilt-node/dist/model/QARuleMatchesRule.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _QARuleMatchesRuleCategory = _interopRequireDefault(require("./QARuleMatchesRuleCategory"));

var _QARuleMatchesRuleUrls = _interopRequireDefault(require("./QARuleMatchesRuleUrls"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The QARuleMatchesRule model module.
 * @module model/QARuleMatchesRule
 * @version 0.5.0
 */
var QARuleMatchesRule = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>QARuleMatchesRule</code>.
   * @alias module:model/QARuleMatchesRule
   * @param category {module:model/QARuleMatchesRuleCategory} 
   * @param description {String} 
   * @param id {String} An rule's identifier that's unique for this language.
   */
  function QARuleMatchesRule(category, description, id) {
    _classCallCheck(this, QARuleMatchesRule);

    QARuleMatchesRule.initialize(this, category, description, id);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(QARuleMatchesRule, null, [{
    key: "initialize",
    value: function initialize(obj, category, description, id) {
      obj['category'] = category;
      obj['description'] = description;
      obj['id'] = id;
    }
    /**
     * Constructs a <code>QARuleMatchesRule</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/QARuleMatchesRule} obj Optional instance to populate.
     * @return {module:model/QARuleMatchesRule} The populated <code>QARuleMatchesRule</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new QARuleMatchesRule();

        if (data.hasOwnProperty('category')) {
          obj['category'] = _QARuleMatchesRuleCategory["default"].constructFromObject(data['category']);
        }

        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('issueType')) {
          obj['issueType'] = _ApiClient["default"].convertToType(data['issueType'], 'String');
        }

        if (data.hasOwnProperty('subId')) {
          obj['subId'] = _ApiClient["default"].convertToType(data['subId'], 'String');
        }

        if (data.hasOwnProperty('urls')) {
          obj['urls'] = _ApiClient["default"].convertToType(data['urls'], [_QARuleMatchesRuleUrls["default"]]);
        }
      }

      return obj;
    }
  }]);

  return QARuleMatchesRule;
}();
/**
 * @member {module:model/QARuleMatchesRuleCategory} category
 */


QARuleMatchesRule.prototype['category'] = undefined;
/**
 * @member {String} description
 */

QARuleMatchesRule.prototype['description'] = undefined;
/**
 * An rule's identifier that's unique for this language.
 * @member {String} id
 */

QARuleMatchesRule.prototype['id'] = undefined;
/**
 * The Localization Quality Issue Type. This is not defined for all languages, in which case it will always be 'Uncategorized'.
 * @member {String} issueType
 */

QARuleMatchesRule.prototype['issueType'] = undefined;
/**
 * An optional sub identifier of the rule, used when several rules are grouped.
 * @member {String} subId
 */

QARuleMatchesRule.prototype['subId'] = undefined;
/**
 * An optional array of URLs with a more detailed description of the error.
 * @member {Array.<module:model/QARuleMatchesRuleUrls>} urls
 */

QARuleMatchesRule.prototype['urls'] = undefined;
var _default = QARuleMatchesRule;
exports["default"] = _default;
