// lilt-node/dist/model/QARuleMatchesMatches 2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _QARuleMatchesContext = _interopRequireDefault(require("./QARuleMatchesContext"));

var _QARuleMatchesReplacements = _interopRequireDefault(require("./QARuleMatchesReplacements"));

var _QARuleMatchesRule = _interopRequireDefault(require("./QARuleMatchesRule"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The QARuleMatchesMatches model module.
 * @module model/QARuleMatchesMatches
 * @version 0.5.0
 */
var QARuleMatchesMatches = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>QARuleMatchesMatches</code>.
   * @alias module:model/QARuleMatchesMatches
   * @param context {module:model/QARuleMatchesContext} 
   * @param length {Number} The length of the error in characters.
   * @param message {String} Message about the error displayed to the user.
   * @param offset {Number} The 0-based character offset of the error in the text.
   * @param replacements {Array.<module:model/QARuleMatchesReplacements>} Replacements that might correct the error. The array can be empty, in this case there is no suggested replacement.
   */
  function QARuleMatchesMatches(context, length, message, offset, replacements) {
    _classCallCheck(this, QARuleMatchesMatches);

    QARuleMatchesMatches.initialize(this, context, length, message, offset, replacements);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(QARuleMatchesMatches, null, [{
    key: "initialize",
    value: function initialize(obj, context, length, message, offset, replacements) {
      obj['context'] = context;
      obj['length'] = length;
      obj['message'] = message;
      obj['offset'] = offset;
      obj['replacements'] = replacements;
    }
    /**
     * Constructs a <code>QARuleMatchesMatches</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/QARuleMatchesMatches} obj Optional instance to populate.
     * @return {module:model/QARuleMatchesMatches} The populated <code>QARuleMatchesMatches</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new QARuleMatchesMatches();

        if (data.hasOwnProperty('context')) {
          obj['context'] = _QARuleMatchesContext["default"].constructFromObject(data['context']);
        }

        if (data.hasOwnProperty('length')) {
          obj['length'] = _ApiClient["default"].convertToType(data['length'], 'Number');
        }

        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], 'String');
        }

        if (data.hasOwnProperty('offset')) {
          obj['offset'] = _ApiClient["default"].convertToType(data['offset'], 'Number');
        }

        if (data.hasOwnProperty('replacements')) {
          obj['replacements'] = _ApiClient["default"].convertToType(data['replacements'], [_QARuleMatchesReplacements["default"]]);
        }

        if (data.hasOwnProperty('rule')) {
          obj['rule'] = _QARuleMatchesRule["default"].constructFromObject(data['rule']);
        }

        if (data.hasOwnProperty('shortMessage')) {
          obj['shortMessage'] = _ApiClient["default"].convertToType(data['shortMessage'], 'String');
        }
      }

      return obj;
    }
  }]);

  return QARuleMatchesMatches;
}();
/**
 * @member {module:model/QARuleMatchesContext} context
 */


QARuleMatchesMatches.prototype['context'] = undefined;
/**
 * The length of the error in characters.
 * @member {Number} length
 */

QARuleMatchesMatches.prototype['length'] = undefined;
/**
 * Message about the error displayed to the user.
 * @member {String} message
 */

QARuleMatchesMatches.prototype['message'] = undefined;
/**
 * The 0-based character offset of the error in the text.
 * @member {Number} offset
 */

QARuleMatchesMatches.prototype['offset'] = undefined;
/**
 * Replacements that might correct the error. The array can be empty, in this case there is no suggested replacement.
 * @member {Array.<module:model/QARuleMatchesReplacements>} replacements
 */

QARuleMatchesMatches.prototype['replacements'] = undefined;
/**
 * @member {module:model/QARuleMatchesRule} rule
 */

QARuleMatchesMatches.prototype['rule'] = undefined;
/**
 * An optional shorter version of 'message'.
 * @member {String} shortMessage
 */

QARuleMatchesMatches.prototype['shortMessage'] = undefined;
var _default = QARuleMatchesMatches;
exports["default"] = _default;
