// lilt-node/dist/model/ResourceStatus.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ResourceStatus model module.
 * @module model/ResourceStatus
 * @version 0.5.0
 */
var ResourceStatus = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ResourceStatus</code>.
   * The status of a resource working on a Project. 
   * @alias module:model/ResourceStatus
   */
  function ResourceStatus() {
    _classCallCheck(this, ResourceStatus);

    ResourceStatus.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ResourceStatus, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ResourceStatus</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ResourceStatus} obj Optional instance to populate.
     * @return {module:model/ResourceStatus} The populated <code>ResourceStatus</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ResourceStatus();

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('num_words_confirmed')) {
          obj['num_words_confirmed'] = _ApiClient["default"].convertToType(data['num_words_confirmed'], 'Number');
        }

        if (data.hasOwnProperty('num_words_new')) {
          obj['num_words_new'] = _ApiClient["default"].convertToType(data['num_words_new'], 'Number');
        }

        if (data.hasOwnProperty('num_words_fuzzy')) {
          obj['num_words_fuzzy'] = _ApiClient["default"].convertToType(data['num_words_fuzzy'], 'Number');
        }

        if (data.hasOwnProperty('num_words_exact')) {
          obj['num_words_exact'] = _ApiClient["default"].convertToType(data['num_words_exact'], 'Number');
        }

        if (data.hasOwnProperty('num_words_reviewed')) {
          obj['num_words_reviewed'] = _ApiClient["default"].convertToType(data['num_words_reviewed'], 'Number');
        }

        if (data.hasOwnProperty('time_elapsed')) {
          obj['time_elapsed'] = _ApiClient["default"].convertToType(data['time_elapsed'], 'Number');
        }

        if (data.hasOwnProperty('time_elapsed_translation')) {
          obj['time_elapsed_translation'] = _ApiClient["default"].convertToType(data['time_elapsed_translation'], 'Number');
        }

        if (data.hasOwnProperty('time_elapsed_research')) {
          obj['time_elapsed_research'] = _ApiClient["default"].convertToType(data['time_elapsed_research'], 'Number');
        }

        if (data.hasOwnProperty('time_elapsed_review')) {
          obj['time_elapsed_review'] = _ApiClient["default"].convertToType(data['time_elapsed_review'], 'Number');
        }

        if (data.hasOwnProperty('num_segments_confirmed')) {
          obj['num_segments_confirmed'] = _ApiClient["default"].convertToType(data['num_segments_confirmed'], 'Number');
        }

        if (data.hasOwnProperty('num_segments_reviewed')) {
          obj['num_segments_reviewed'] = _ApiClient["default"].convertToType(data['num_segments_reviewed'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return ResourceStatus;
}();
/**
 * An email address.
 * @member {String} email
 */


ResourceStatus.prototype['email'] = undefined;
/**
 * The full name.
 * @member {String} name
 */

ResourceStatus.prototype['name'] = undefined;
/**
 * The number of source words confirmed.
 * @member {Number} num_words_confirmed
 */

ResourceStatus.prototype['num_words_confirmed'] = undefined;
/**
 * The number of new source words confirmed.
 * @member {Number} num_words_new
 */

ResourceStatus.prototype['num_words_new'] = undefined;
/**
 * The number of fuzzy match source words confirmed.
 * @member {Number} num_words_fuzzy
 */

ResourceStatus.prototype['num_words_fuzzy'] = undefined;
/**
 * The number of exact match source words confirmed.
 * @member {Number} num_words_exact
 */

ResourceStatus.prototype['num_words_exact'] = undefined;
/**
 * The number of source words reviewed.
 * @member {Number} num_words_reviewed
 */

ResourceStatus.prototype['num_words_reviewed'] = undefined;
/**
 * The total time spent on translation and research. Measured in milliseconds.
 * @member {Number} time_elapsed
 */

ResourceStatus.prototype['time_elapsed'] = undefined;
/**
 * The total time spent translating. Measured in milliseconds.
 * @member {Number} time_elapsed_translation
 */

ResourceStatus.prototype['time_elapsed_translation'] = undefined;
/**
 * The total time spent on research. Measured in milliseconds.
 * @member {Number} time_elapsed_research
 */

ResourceStatus.prototype['time_elapsed_research'] = undefined;
/**
 * The total time spent reviewing. Measured in milliseconds.
 * @member {Number} time_elapsed_review
 */

ResourceStatus.prototype['time_elapsed_review'] = undefined;
/**
 * The number of source segments confirmed.
 * @member {Number} num_segments_confirmed
 */

ResourceStatus.prototype['num_segments_confirmed'] = undefined;
/**
 * The number of source segments reviewed.
 * @member {Number} num_segments_reviewed
 */

ResourceStatus.prototype['num_segments_reviewed'] = undefined;
var _default = ResourceStatus;
exports["default"] = _default;
