// lilt-node/dist/model/Segment 2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Segment model module.
 * @module model/Segment
 * @version 0.5.0
 */
var Segment = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Segment</code>.
   * A Segment is a source string and, optionally, its translation. A Segment can be associated with both a Memory and a Document. The Segment object contains additional metadata about the source and target strings. 
   * @alias module:model/Segment
   */
  function Segment() {
    _classCallCheck(this, Segment);

    Segment.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Segment, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Segment</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Segment} obj Optional instance to populate.
     * @return {module:model/Segment} The populated <code>Segment</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Segment();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }

        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Number');
        }

        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Number');
        }

        if (data.hasOwnProperty('document_id')) {
          obj['document_id'] = _ApiClient["default"].convertToType(data['document_id'], 'Number');
        }

        if (data.hasOwnProperty('memory_id')) {
          obj['memory_id'] = _ApiClient["default"].convertToType(data['memory_id'], 'Number');
        }

        if (data.hasOwnProperty('source')) {
          obj['source'] = _ApiClient["default"].convertToType(data['source'], 'String');
        }

        if (data.hasOwnProperty('srclang')) {
          obj['srclang'] = _ApiClient["default"].convertToType(data['srclang'], 'String');
        }

        if (data.hasOwnProperty('target')) {
          obj['target'] = _ApiClient["default"].convertToType(data['target'], 'String');
        }

        if (data.hasOwnProperty('trglang')) {
          obj['trglang'] = _ApiClient["default"].convertToType(data['trglang'], 'String');
        }

        if (data.hasOwnProperty('is_confirmed')) {
          obj['is_confirmed'] = _ApiClient["default"].convertToType(data['is_confirmed'], 'Boolean');
        }

        if (data.hasOwnProperty('is_reviewed')) {
          obj['is_reviewed'] = _ApiClient["default"].convertToType(data['is_reviewed'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return Segment;
}();
/**
 * A unique number identifying the Segment.
 * @member {Number} id
 */


Segment.prototype['id'] = undefined;
/**
 * Time at which the object was created. Measured in seconds since the Unix epoch.
 * @member {Number} created_at
 */

Segment.prototype['created_at'] = undefined;
/**
 * Time at which the object was created. Measured in seconds since the Unix epoch.
 * @member {Number} updated_at
 */

Segment.prototype['updated_at'] = undefined;
/**
 * A unique Document identifier.
 * @member {Number} document_id
 */

Segment.prototype['document_id'] = undefined;
/**
 * The Memory with which this Segment is associated.
 * @member {Number} memory_id
 */

Segment.prototype['memory_id'] = undefined;
/**
 * The source string.
 * @member {String} source
 */

Segment.prototype['source'] = undefined;
/**
 * An ISO 639-1 language code.
 * @member {String} srclang
 */

Segment.prototype['srclang'] = undefined;
/**
 * The target string.
 * @member {String} target
 */

Segment.prototype['target'] = undefined;
/**
 * An ISO 639-1 language code.
 * @member {String} trglang
 */

Segment.prototype['trglang'] = undefined;
/**
 * The confirmation status.
 * @member {Boolean} is_confirmed
 */

Segment.prototype['is_confirmed'] = undefined;
/**
 * The review status.
 * @member {Boolean} is_reviewed
 */

Segment.prototype['is_reviewed'] = undefined;
var _default = Segment;
exports["default"] = _default;
