// lilt-node/dist/model/SegmentCreateParameters.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SegmentCreateParameters model module.
 * @module model/SegmentCreateParameters
 * @version 0.5.0
 */
var SegmentCreateParameters = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SegmentCreateParameters</code>.
   * @alias module:model/SegmentCreateParameters
   * @param source {String} The source string.
   */
  function SegmentCreateParameters(source) {
    _classCallCheck(this, SegmentCreateParameters);

    SegmentCreateParameters.initialize(this, source);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SegmentCreateParameters, null, [{
    key: "initialize",
    value: function initialize(obj, source) {
      obj['source'] = source;
    }
    /**
     * Constructs a <code>SegmentCreateParameters</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SegmentCreateParameters} obj Optional instance to populate.
     * @return {module:model/SegmentCreateParameters} The populated <code>SegmentCreateParameters</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SegmentCreateParameters();

        if (data.hasOwnProperty('memory_id')) {
          obj['memory_id'] = _ApiClient["default"].convertToType(data['memory_id'], 'Number');
        }

        if (data.hasOwnProperty('document_id')) {
          obj['document_id'] = _ApiClient["default"].convertToType(data['document_id'], 'Number');
        }

        if (data.hasOwnProperty('source')) {
          obj['source'] = _ApiClient["default"].convertToType(data['source'], 'String');
        }

        if (data.hasOwnProperty('target')) {
          obj['target'] = _ApiClient["default"].convertToType(data['target'], 'String');
        }
      }

      return obj;
    }
  }]);

  return SegmentCreateParameters;
}();
/**
 * A unique Memory identifier.
 * @member {Number} memory_id
 */


SegmentCreateParameters.prototype['memory_id'] = undefined;
/**
 * A unique Document identifier.
 * @member {Number} document_id
 */

SegmentCreateParameters.prototype['document_id'] = undefined;
/**
 * The source string.
 * @member {String} source
 */

SegmentCreateParameters.prototype['source'] = undefined;
/**
 * The target string.
 * @member {String} target
 */

SegmentCreateParameters.prototype['target'] = undefined;
var _default = SegmentCreateParameters;
exports["default"] = _default;
