// lilt-node/dist/model/TranslationList.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Translation = _interopRequireDefault(require("./Translation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TranslationList model module.
 * @module model/TranslationList
 * @version 0.5.0
 */
var TranslationList = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TranslationList</code>.
   * An ranked list of translations and associated metadata.
   * @alias module:model/TranslationList
   */
  function TranslationList() {
    _classCallCheck(this, TranslationList);

    TranslationList.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TranslationList, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>TranslationList</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TranslationList} obj Optional instance to populate.
     * @return {module:model/TranslationList} The populated <code>TranslationList</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TranslationList();

        if (data.hasOwnProperty('untokenizedSource')) {
          obj['untokenizedSource'] = _ApiClient["default"].convertToType(data['untokenizedSource'], 'String');
        }

        if (data.hasOwnProperty('tokenizedSource')) {
          obj['tokenizedSource'] = _ApiClient["default"].convertToType(data['tokenizedSource'], 'String');
        }

        if (data.hasOwnProperty('sourceDelimiters')) {
          obj['sourceDelimiters'] = _ApiClient["default"].convertToType(data['sourceDelimiters'], ['String']);
        }

        if (data.hasOwnProperty('translation')) {
          obj['translation'] = _ApiClient["default"].convertToType(data['translation'], [_Translation["default"]]);
        }
      }

      return obj;
    }
  }]);

  return TranslationList;
}();
/**
 * The untokenized source segment. Punctuation has not been separated from words.
 * @member {String} untokenizedSource
 */


TranslationList.prototype['untokenizedSource'] = undefined;
/**
 * The tokenized source segment. Punctuation has been separated from words.
 * @member {String} tokenizedSource
 */

TranslationList.prototype['tokenizedSource'] = undefined;
/**
 * A format string that indicates, for each word, if the word should be preceded by a space.
 * @member {Array.<String>} sourceDelimiters
 */

TranslationList.prototype['sourceDelimiters'] = undefined;
/**
 * A list of Translation objects.
 * @member {Array.<module:model/Translation>} translation
 */

TranslationList.prototype['translation'] = undefined;
var _default = TranslationList;
exports["default"] = _default;
