// lilt-node/dist/model/TranslationMemoryEntry 2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TranslationMemoryEntry model module.
 * @module model/TranslationMemoryEntry
 * @version 0.5.0
 */
var TranslationMemoryEntry = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TranslationMemoryEntry</code>.
   * A translation memory entry.
   * @alias module:model/TranslationMemoryEntry
   */
  function TranslationMemoryEntry() {
    _classCallCheck(this, TranslationMemoryEntry);

    TranslationMemoryEntry.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TranslationMemoryEntry, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>TranslationMemoryEntry</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TranslationMemoryEntry} obj Optional instance to populate.
     * @return {module:model/TranslationMemoryEntry} The populated <code>TranslationMemoryEntry</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TranslationMemoryEntry();

        if (data.hasOwnProperty('source')) {
          obj['source'] = _ApiClient["default"].convertToType(data['source'], 'String');
        }

        if (data.hasOwnProperty('target')) {
          obj['target'] = _ApiClient["default"].convertToType(data['target'], 'String');
        }

        if (data.hasOwnProperty('score')) {
          obj['score'] = _ApiClient["default"].convertToType(data['score'], 'Number');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], Object);
        }
      }

      return obj;
    }
  }]);

  return TranslationMemoryEntry;
}();
/**
 * The source string.
 * @member {String} source
 */


TranslationMemoryEntry.prototype['source'] = undefined;
/**
 * The target string. Tags will be automatically placed according to the query string.
 * @member {String} target
 */

TranslationMemoryEntry.prototype['target'] = undefined;
/**
 * The fuzzy match score.
 * @member {Number} score
 */

TranslationMemoryEntry.prototype['score'] = undefined;
/**
 * Attributes describing the translation memory entry.
 * @member {Object} metadata
 */

TranslationMemoryEntry.prototype['metadata'] = undefined;
var _default = TranslationMemoryEntry;
exports["default"] = _default;
