// lilt-node/dist/api/TranslateApi 2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Error = _interopRequireDefault(require("../model/Error"));

var _TranslateRegisterResponse = _interopRequireDefault(require("../model/TranslateRegisterResponse"));

var _TranslationList = _interopRequireDefault(require("../model/TranslationList"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
* Translate service.
* @module api/TranslateApi
* @version 0.5.0
*/
var TranslateApi = /*#__PURE__*/function () {
  /**
  * Constructs a new TranslateApi. 
  * @alias module:api/TranslateApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function TranslateApi(apiClient) {
    _classCallCheck(this, TranslateApi);

    this.apiClient = apiClient || _ApiClient["default"].instance;
  }
  /**
   * Register a segment
   * Register a source string for interactive translation. The `source_hash` value that is returned by this request is required by the `prefix` parameter for the translation endpoint. The maximum source length is 5,000 characters. Usage charges apply to this endpoint for production REST API keys.  
   * @param {String} source A source string to be registered.
   * @param {String} srclang An ISO 639-1 language code.
   * @param {String} trglang An ISO 639-1 language code.
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/TranslateRegisterResponse} and HTTP response
   */


  _createClass(TranslateApi, [{
    key: "registerSegmentWithHttpInfo",
    value: function registerSegmentWithHttpInfo(source, srclang, trglang) {
      var postBody = null; // verify the required parameter 'source' is set

      if (source === undefined || source === null) {
        throw new _Error["default"]("Missing the required parameter 'source' when calling registerSegment");
      } // verify the required parameter 'srclang' is set


      if (srclang === undefined || srclang === null) {
        throw new _Error["default"]("Missing the required parameter 'srclang' when calling registerSegment");
      } // verify the required parameter 'trglang' is set


      if (trglang === undefined || trglang === null) {
        throw new _Error["default"]("Missing the required parameter 'trglang' when calling registerSegment");
      }

      var pathParams = {};
      var queryParams = {
        'source': source,
        'srclang': srclang,
        'trglang': trglang
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['ApiKeyAuth', 'BasicAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _TranslateRegisterResponse["default"];
      return this.apiClient.callApi('/translate/register', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }
    /**
     * Register a segment
     * Register a source string for interactive translation. The `source_hash` value that is returned by this request is required by the `prefix` parameter for the translation endpoint. The maximum source length is 5,000 characters. Usage charges apply to this endpoint for production REST API keys.  
     * @param {String} source A source string to be registered.
     * @param {String} srclang An ISO 639-1 language code.
     * @param {String} trglang An ISO 639-1 language code.
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/TranslateRegisterResponse}
     */

  }, {
    key: "registerSegment",
    value: function registerSegment(source, srclang, trglang) {
      return this.registerSegmentWithHttpInfo(source, srclang, trglang).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
    /**
     * Translate a segment
     * Translate a source string.  Setting the `rich` parameter to `true` will change the response format to include additional information about each translation including a model score, word alignments,  and formatting information. The rich format can be seen in the example response on this page.  By default, this endpoint also returns translation memory (TM) fuzzy matches, along with associated scores. Fuzzy matches always appear ahead of machine translation output in the response.  The maximum source length is 5,000 characters.  Usage charges apply to this endpoint for production REST API keys.  
     * @param {Number} memoryId A unique Memory identifier.
     * @param {Object} opts Optional parameters
     * @param {String} opts.source The source text to be translated.
     * @param {Number} opts.sourceHash A source hash code.
     * @param {String} opts.prefix A target prefix.
     * @param {Number} opts.n Return top n translations (deprecated). (default to 1)
     * @param {Boolean} opts.rich Returns rich translation information (e.g., with word alignments). (default to false)
     * @param {Boolean} opts.tmMatches Include translation memory fuzzy matches. (default to true)
     * @param {Boolean} opts.projectTags Project tags. Projects tags in source to target if set to true. (default to false)
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/TranslationList} and HTTP response
     */

  }, {
    key: "translateSegmentWithHttpInfo",
    value: function translateSegmentWithHttpInfo(memoryId, opts) {
      opts = opts || {};
      var postBody = null; // verify the required parameter 'memoryId' is set

      if (memoryId === undefined || memoryId === null) {
        throw new _Error["default"]("Missing the required parameter 'memoryId' when calling translateSegment");
      }

      var pathParams = {};
      var queryParams = {
        'memory_id': memoryId,
        'source': opts['source'],
        'source_hash': opts['sourceHash'],
        'prefix': opts['prefix'],
        'n': opts['n'],
        'rich': opts['rich'],
        'tm_matches': opts['tmMatches'],
        'project_tags': opts['projectTags']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['ApiKeyAuth', 'BasicAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _TranslationList["default"];
      return this.apiClient.callApi('/translate', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }
    /**
     * Translate a segment
     * Translate a source string.  Setting the `rich` parameter to `true` will change the response format to include additional information about each translation including a model score, word alignments,  and formatting information. The rich format can be seen in the example response on this page.  By default, this endpoint also returns translation memory (TM) fuzzy matches, along with associated scores. Fuzzy matches always appear ahead of machine translation output in the response.  The maximum source length is 5,000 characters.  Usage charges apply to this endpoint for production REST API keys.  
     * @param {Number} memoryId A unique Memory identifier.
     * @param {Object} opts Optional parameters
     * @param {String} opts.source The source text to be translated.
     * @param {Number} opts.sourceHash A source hash code.
     * @param {String} opts.prefix A target prefix.
     * @param {Number} opts.n Return top n translations (deprecated). (default to 1)
     * @param {Boolean} opts.rich Returns rich translation information (e.g., with word alignments). (default to false)
     * @param {Boolean} opts.tmMatches Include translation memory fuzzy matches. (default to true)
     * @param {Boolean} opts.projectTags Project tags. Projects tags in source to target if set to true. (default to false)
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/TranslationList}
     */

  }, {
    key: "translateSegment",
    value: function translateSegment(memoryId, opts) {
      return this.translateSegmentWithHttpInfo(memoryId, opts).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
  }]);

  return TranslateApi;
}();

exports["default"] = TranslateApi;
