// lilt-node/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient["default"];
  }
});
Object.defineProperty(exports, "Annotation", {
  enumerable: true,
  get: function get() {
    return _Annotation["default"];
  }
});
Object.defineProperty(exports, "ApiRoot", {
  enumerable: true,
  get: function get() {
    return _ApiRoot["default"];
  }
});
Object.defineProperty(exports, "Comment", {
  enumerable: true,
  get: function get() {
    return _Comment["default"];
  }
});
Object.defineProperty(exports, "Connector", {
  enumerable: true,
  get: function get() {
    return _Connector["default"];
  }
});
Object.defineProperty(exports, "ConnectorArguments", {
  enumerable: true,
  get: function get() {
    return _ConnectorArguments["default"];
  }
});
Object.defineProperty(exports, "ConnectorDeleteResponse", {
  enumerable: true,
  get: function get() {
    return _ConnectorDeleteResponse["default"];
  }
});
Object.defineProperty(exports, "DocumentAssignmentParameters", {
  enumerable: true,
  get: function get() {
    return _DocumentAssignmentParameters["default"];
  }
});
Object.defineProperty(exports, "DocumentAssignmentResponse", {
  enumerable: true,
  get: function get() {
    return _DocumentAssignmentResponse["default"];
  }
});
Object.defineProperty(exports, "DocumentDeleteResponse", {
  enumerable: true,
  get: function get() {
    return _DocumentDeleteResponse["default"];
  }
});
Object.defineProperty(exports, "DocumentParameters", {
  enumerable: true,
  get: function get() {
    return _DocumentParameters["default"];
  }
});
Object.defineProperty(exports, "DocumentPretranslateParameters", {
  enumerable: true,
  get: function get() {
    return _DocumentPretranslateParameters["default"];
  }
});
Object.defineProperty(exports, "DocumentPretranslateResponse", {
  enumerable: true,
  get: function get() {
    return _DocumentPretranslateResponse["default"];
  }
});
Object.defineProperty(exports, "DocumentPretranslating", {
  enumerable: true,
  get: function get() {
    return _DocumentPretranslating["default"];
  }
});
Object.defineProperty(exports, "DocumentPretranslatingStatus", {
  enumerable: true,
  get: function get() {
    return _DocumentPretranslatingStatus["default"];
  }
});
Object.defineProperty(exports, "DocumentQuote", {
  enumerable: true,
  get: function get() {
    return _DocumentQuote["default"];
  }
});
Object.defineProperty(exports, "DocumentUpdateParameters", {
  enumerable: true,
  get: function get() {
    return _DocumentUpdateParameters["default"];
  }
});
Object.defineProperty(exports, "DocumentWithSegments", {
  enumerable: true,
  get: function get() {
    return _DocumentWithSegments["default"];
  }
});
Object.defineProperty(exports, "DocumentWithoutSegments", {
  enumerable: true,
  get: function get() {
    return _DocumentWithoutSegments["default"];
  }
});
Object.defineProperty(exports, "DocumentWithoutSegmentsStatus", {
  enumerable: true,
  get: function get() {
    return _DocumentWithoutSegmentsStatus["default"];
  }
});
Object.defineProperty(exports, "Error", {
  enumerable: true,
  get: function get() {
    return _Error["default"];
  }
});
Object.defineProperty(exports, "File", {
  enumerable: true,
  get: function get() {
    return _File["default"];
  }
});
Object.defineProperty(exports, "FileDeleteResponse", {
  enumerable: true,
  get: function get() {
    return _FileDeleteResponse["default"];
  }
});
Object.defineProperty(exports, "LanguagesResponse", {
  enumerable: true,
  get: function get() {
    return _LanguagesResponse["default"];
  }
});
Object.defineProperty(exports, "LexiconEntry", {
  enumerable: true,
  get: function get() {
    return _LexiconEntry["default"];
  }
});
Object.defineProperty(exports, "LexiconEntryExamples", {
  enumerable: true,
  get: function get() {
    return _LexiconEntryExamples["default"];
  }
});
Object.defineProperty(exports, "LexiconEntrySourceSpan", {
  enumerable: true,
  get: function get() {
    return _LexiconEntrySourceSpan["default"];
  }
});
Object.defineProperty(exports, "LexiconEntryTargetSpan", {
  enumerable: true,
  get: function get() {
    return _LexiconEntryTargetSpan["default"];
  }
});
Object.defineProperty(exports, "LexiconEntryTranslations", {
  enumerable: true,
  get: function get() {
    return _LexiconEntryTranslations["default"];
  }
});
Object.defineProperty(exports, "LexiconUpdateParameters", {
  enumerable: true,
  get: function get() {
    return _LexiconUpdateParameters["default"];
  }
});
Object.defineProperty(exports, "LexiconUpdateResponse", {
  enumerable: true,
  get: function get() {
    return _LexiconUpdateResponse["default"];
  }
});
Object.defineProperty(exports, "MatchBand", {
  enumerable: true,
  get: function get() {
    return _MatchBand["default"];
  }
});
Object.defineProperty(exports, "Memory", {
  enumerable: true,
  get: function get() {
    return _Memory["default"];
  }
});
Object.defineProperty(exports, "MemoryCreateParameters", {
  enumerable: true,
  get: function get() {
    return _MemoryCreateParameters["default"];
  }
});
Object.defineProperty(exports, "MemoryDeleteResponse", {
  enumerable: true,
  get: function get() {
    return _MemoryDeleteResponse["default"];
  }
});
Object.defineProperty(exports, "MemoryImportResponse", {
  enumerable: true,
  get: function get() {
    return _MemoryImportResponse["default"];
  }
});
Object.defineProperty(exports, "MemoryInsertResponse", {
  enumerable: true,
  get: function get() {
    return _MemoryInsertResponse["default"];
  }
});
Object.defineProperty(exports, "MemoryUpdateParameters", {
  enumerable: true,
  get: function get() {
    return _MemoryUpdateParameters["default"];
  }
});
Object.defineProperty(exports, "MemoryUpdateResponse", {
  enumerable: true,
  get: function get() {
    return _MemoryUpdateResponse["default"];
  }
});
Object.defineProperty(exports, "Project", {
  enumerable: true,
  get: function get() {
    return _Project["default"];
  }
});
Object.defineProperty(exports, "ProjectCreateParameters", {
  enumerable: true,
  get: function get() {
    return _ProjectCreateParameters["default"];
  }
});
Object.defineProperty(exports, "ProjectDeleteResponse", {
  enumerable: true,
  get: function get() {
    return _ProjectDeleteResponse["default"];
  }
});
Object.defineProperty(exports, "ProjectQuote", {
  enumerable: true,
  get: function get() {
    return _ProjectQuote["default"];
  }
});
Object.defineProperty(exports, "ProjectStatus", {
  enumerable: true,
  get: function get() {
    return _ProjectStatus["default"];
  }
});
Object.defineProperty(exports, "ProjectUpdateResponse", {
  enumerable: true,
  get: function get() {
    return _ProjectUpdateResponse["default"];
  }
});
Object.defineProperty(exports, "QARuleMatches", {
  enumerable: true,
  get: function get() {
    return _QARuleMatches["default"];
  }
});
Object.defineProperty(exports, "QARuleMatchesContext", {
  enumerable: true,
  get: function get() {
    return _QARuleMatchesContext["default"];
  }
});
Object.defineProperty(exports, "QARuleMatchesMatches", {
  enumerable: true,
  get: function get() {
    return _QARuleMatchesMatches["default"];
  }
});
Object.defineProperty(exports, "QARuleMatchesReplacements", {
  enumerable: true,
  get: function get() {
    return _QARuleMatchesReplacements["default"];
  }
});
Object.defineProperty(exports, "QARuleMatchesRule", {
  enumerable: true,
  get: function get() {
    return _QARuleMatchesRule["default"];
  }
});
Object.defineProperty(exports, "QARuleMatchesRuleCategory", {
  enumerable: true,
  get: function get() {
    return _QARuleMatchesRuleCategory["default"];
  }
});
Object.defineProperty(exports, "QARuleMatchesRuleUrls", {
  enumerable: true,
  get: function get() {
    return _QARuleMatchesRuleUrls["default"];
  }
});
Object.defineProperty(exports, "ResourceStatus", {
  enumerable: true,
  get: function get() {
    return _ResourceStatus["default"];
  }
});
Object.defineProperty(exports, "Segment", {
  enumerable: true,
  get: function get() {
    return _Segment["default"];
  }
});
Object.defineProperty(exports, "SegmentCreateParameters", {
  enumerable: true,
  get: function get() {
    return _SegmentCreateParameters["default"];
  }
});
Object.defineProperty(exports, "SegmentDeleteResponse", {
  enumerable: true,
  get: function get() {
    return _SegmentDeleteResponse["default"];
  }
});
Object.defineProperty(exports, "SegmentUpdateParameters", {
  enumerable: true,
  get: function get() {
    return _SegmentUpdateParameters["default"];
  }
});
Object.defineProperty(exports, "SegmentWithComments", {
  enumerable: true,
  get: function get() {
    return _SegmentWithComments["default"];
  }
});
Object.defineProperty(exports, "TaggedSegment", {
  enumerable: true,
  get: function get() {
    return _TaggedSegment["default"];
  }
});
Object.defineProperty(exports, "TranslateRegisterResponse", {
  enumerable: true,
  get: function get() {
    return _TranslateRegisterResponse["default"];
  }
});
Object.defineProperty(exports, "Translation", {
  enumerable: true,
  get: function get() {
    return _Translation["default"];
  }
});
Object.defineProperty(exports, "TranslationList", {
  enumerable: true,
  get: function get() {
    return _TranslationList["default"];
  }
});
Object.defineProperty(exports, "TranslationMemoryEntry", {
  enumerable: true,
  get: function get() {
    return _TranslationMemoryEntry["default"];
  }
});
Object.defineProperty(exports, "ConnectorsApi", {
  enumerable: true,
  get: function get() {
    return _ConnectorsApi["default"];
  }
});
Object.defineProperty(exports, "DocumentsApi", {
  enumerable: true,
  get: function get() {
    return _DocumentsApi["default"];
  }
});
Object.defineProperty(exports, "FilesApi", {
  enumerable: true,
  get: function get() {
    return _FilesApi["default"];
  }
});
Object.defineProperty(exports, "LanguagesApi", {
  enumerable: true,
  get: function get() {
    return _LanguagesApi["default"];
  }
});
Object.defineProperty(exports, "LexiconApi", {
  enumerable: true,
  get: function get() {
    return _LexiconApi["default"];
  }
});
Object.defineProperty(exports, "MemoriesApi", {
  enumerable: true,
  get: function get() {
    return _MemoriesApi["default"];
  }
});
Object.defineProperty(exports, "ProjectsApi", {
  enumerable: true,
  get: function get() {
    return _ProjectsApi["default"];
  }
});
Object.defineProperty(exports, "QAApi", {
  enumerable: true,
  get: function get() {
    return _QAApi["default"];
  }
});
Object.defineProperty(exports, "RootApi", {
  enumerable: true,
  get: function get() {
    return _RootApi["default"];
  }
});
Object.defineProperty(exports, "SegmentsApi", {
  enumerable: true,
  get: function get() {
    return _SegmentsApi["default"];
  }
});
Object.defineProperty(exports, "TranslateApi", {
  enumerable: true,
  get: function get() {
    return _TranslateApi["default"];
  }
});

var _ApiClient = _interopRequireDefault(require("./ApiClient"));

var _Annotation = _interopRequireDefault(require("./model/Annotation"));

var _ApiRoot = _interopRequireDefault(require("./model/ApiRoot"));

var _Comment = _interopRequireDefault(require("./model/Comment"));

var _Connector = _interopRequireDefault(require("./model/Connector"));

var _ConnectorArguments = _interopRequireDefault(require("./model/ConnectorArguments"));

var _ConnectorDeleteResponse = _interopRequireDefault(require("./model/ConnectorDeleteResponse"));

var _DocumentAssignmentParameters = _interopRequireDefault(require("./model/DocumentAssignmentParameters"));

var _DocumentAssignmentResponse = _interopRequireDefault(require("./model/DocumentAssignmentResponse"));

var _DocumentDeleteResponse = _interopRequireDefault(require("./model/DocumentDeleteResponse"));

var _DocumentParameters = _interopRequireDefault(require("./model/DocumentParameters"));

var _DocumentPretranslateParameters = _interopRequireDefault(require("./model/DocumentPretranslateParameters"));

var _DocumentPretranslateResponse = _interopRequireDefault(require("./model/DocumentPretranslateResponse"));

var _DocumentPretranslating = _interopRequireDefault(require("./model/DocumentPretranslating"));

var _DocumentPretranslatingStatus = _interopRequireDefault(require("./model/DocumentPretranslatingStatus"));

var _DocumentQuote = _interopRequireDefault(require("./model/DocumentQuote"));

var _DocumentUpdateParameters = _interopRequireDefault(require("./model/DocumentUpdateParameters"));

var _DocumentWithSegments = _interopRequireDefault(require("./model/DocumentWithSegments"));

var _DocumentWithoutSegments = _interopRequireDefault(require("./model/DocumentWithoutSegments"));

var _DocumentWithoutSegmentsStatus = _interopRequireDefault(require("./model/DocumentWithoutSegmentsStatus"));

var _Error = _interopRequireDefault(require("./model/Error"));

var _File = _interopRequireDefault(require("./model/File"));

var _FileDeleteResponse = _interopRequireDefault(require("./model/FileDeleteResponse"));

var _LanguagesResponse = _interopRequireDefault(require("./model/LanguagesResponse"));

var _LexiconEntry = _interopRequireDefault(require("./model/LexiconEntry"));

var _LexiconEntryExamples = _interopRequireDefault(require("./model/LexiconEntryExamples"));

var _LexiconEntrySourceSpan = _interopRequireDefault(require("./model/LexiconEntrySourceSpan"));

var _LexiconEntryTargetSpan = _interopRequireDefault(require("./model/LexiconEntryTargetSpan"));

var _LexiconEntryTranslations = _interopRequireDefault(require("./model/LexiconEntryTranslations"));

var _LexiconUpdateParameters = _interopRequireDefault(require("./model/LexiconUpdateParameters"));

var _LexiconUpdateResponse = _interopRequireDefault(require("./model/LexiconUpdateResponse"));

var _MatchBand = _interopRequireDefault(require("./model/MatchBand"));

var _Memory = _interopRequireDefault(require("./model/Memory"));

var _MemoryCreateParameters = _interopRequireDefault(require("./model/MemoryCreateParameters"));

var _MemoryDeleteResponse = _interopRequireDefault(require("./model/MemoryDeleteResponse"));

var _MemoryImportResponse = _interopRequireDefault(require("./model/MemoryImportResponse"));

var _MemoryInsertResponse = _interopRequireDefault(require("./model/MemoryInsertResponse"));

var _MemoryUpdateParameters = _interopRequireDefault(require("./model/MemoryUpdateParameters"));

var _MemoryUpdateResponse = _interopRequireDefault(require("./model/MemoryUpdateResponse"));

var _Project = _interopRequireDefault(require("./model/Project"));

var _ProjectCreateParameters = _interopRequireDefault(require("./model/ProjectCreateParameters"));

var _ProjectDeleteResponse = _interopRequireDefault(require("./model/ProjectDeleteResponse"));

var _ProjectQuote = _interopRequireDefault(require("./model/ProjectQuote"));

var _ProjectStatus = _interopRequireDefault(require("./model/ProjectStatus"));

var _ProjectUpdateResponse = _interopRequireDefault(require("./model/ProjectUpdateResponse"));

var _QARuleMatches = _interopRequireDefault(require("./model/QARuleMatches"));

var _QARuleMatchesContext = _interopRequireDefault(require("./model/QARuleMatchesContext"));

var _QARuleMatchesMatches = _interopRequireDefault(require("./model/QARuleMatchesMatches"));

var _QARuleMatchesReplacements = _interopRequireDefault(require("./model/QARuleMatchesReplacements"));

var _QARuleMatchesRule = _interopRequireDefault(require("./model/QARuleMatchesRule"));

var _QARuleMatchesRuleCategory = _interopRequireDefault(require("./model/QARuleMatchesRuleCategory"));

var _QARuleMatchesRuleUrls = _interopRequireDefault(require("./model/QARuleMatchesRuleUrls"));

var _ResourceStatus = _interopRequireDefault(require("./model/ResourceStatus"));

var _Segment = _interopRequireDefault(require("./model/Segment"));

var _SegmentCreateParameters = _interopRequireDefault(require("./model/SegmentCreateParameters"));

var _SegmentDeleteResponse = _interopRequireDefault(require("./model/SegmentDeleteResponse"));

var _SegmentUpdateParameters = _interopRequireDefault(require("./model/SegmentUpdateParameters"));

var _SegmentWithComments = _interopRequireDefault(require("./model/SegmentWithComments"));

var _TaggedSegment = _interopRequireDefault(require("./model/TaggedSegment"));

var _TranslateRegisterResponse = _interopRequireDefault(require("./model/TranslateRegisterResponse"));

var _Translation = _interopRequireDefault(require("./model/Translation"));

var _TranslationList = _interopRequireDefault(require("./model/TranslationList"));

var _TranslationMemoryEntry = _interopRequireDefault(require("./model/TranslationMemoryEntry"));

var _ConnectorsApi = _interopRequireDefault(require("./api/ConnectorsApi"));

var _DocumentsApi = _interopRequireDefault(require("./api/DocumentsApi"));

var _FilesApi = _interopRequireDefault(require("./api/FilesApi"));

var _LanguagesApi = _interopRequireDefault(require("./api/LanguagesApi"));

var _LexiconApi = _interopRequireDefault(require("./api/LexiconApi"));

var _MemoriesApi = _interopRequireDefault(require("./api/MemoriesApi"));

var _ProjectsApi = _interopRequireDefault(require("./api/ProjectsApi"));

var _QAApi = _interopRequireDefault(require("./api/QAApi"));

var _RootApi = _interopRequireDefault(require("./api/RootApi"));

var _SegmentsApi = _interopRequireDefault(require("./api/SegmentsApi"));

var _TranslateApi = _interopRequireDefault(require("./api/TranslateApi"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
